### DISCUSSION TITLE
Recommendation: you can add AdaFace model to compreface. It is quite good.

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [exadel-inc/CompreFace#879] [Q&A] [answered]
You can check out the AdaFace model to add to compreFace.

**Comment by pospielov** [comment, marked as answer]
The problem with the integration of AdaFace is that originally it was written using the Pytorch framework. We have different builds of CompreFace that uses Tensorflow and MXNet frameworks. 
So it will require a lot of effort to add a new framework. Ideally, somebody should create a repository that standardizes every algorithm, and there was a fantastic attempt: https://github.com/JDAI-CV/FaceX-Zoo. Still, they don't have the models themselves.
There is a serengil/deepface repository, which could be used in CompreFace, but as you mentioned, the accuracy is worse than in CompreFace, and it's hard to say why. Probably they made mistakes when they moved the model from one framework to another.
Finally, there is a zoo gathered by frameworks, like ONNX - https://github.com/onnx/models. But they usually have only a couple of models.
